### DISCUSSION TITLE
Need a bit of help w/ F7Plus

### DISCUSSION THREAD

**Discussion by snhirsch** [keirf/greaseweazle#86] [Q&A]
Hi, All.  Just received my F7 Plus and have started working with it.  The first thing I thought I'd try was reading a plain old PC 360k DSDD diskette to SCP and converting to a sector image.  The F7 is connected to a 1.2M Teac drive.  What I've done:

- gw read --track c=0-39:step=2 test.scp

This cheerfully ticks through all 40 tracks and generates the SCP file.  Then:

- disk-analyse -s 0 -e 39 test.scp test.img

Unfortunately, disk-analyse reports:

``Side 0:
 T0-39: Unformatted*
Side 1:
 T0-39: Unformatted
** WARNING: 40 tracks are damaged or unidentified!
``
Certainly I'm missing something critical here, but what?

**Comment by keirf** [comment]
Specify `--format=ibm_mfm_dd` to disk-analyse. You may also need `--rpm=360:300` (iirc).

**Reply by snhirsch** [reply]
No joy.  The error messages are slightly different, but it still does not extract any data from the bitstream.

**Reply by keirf** [reply]
If you can make your SCP dump available, I'll have a look.

**Reply by snhirsch** [reply]
Here you go!
[test.scp.gz](https://github.com/keirf/Greaseweazle/files/6071820/test.scp.gz)

**Reply by keirf** [reply]
The flux is a blurry mess. Possibly this has been read in HD mode rather than DD mode. Try `gw pin 2 L` before the read command. Pin 2 is often used to tell a 5.25 drive what density it is operating at.

**Reply by snhirsch** [reply]
Turns out this diskette is SS, so I sent it 'gw pin 2 L' followed by:

gw read --track c=0-39:h=0:step=2 --revs=5 test.scp

[test.scp.gz](https://github.com/keirf/Greaseweazle/files/6072310/test.scp.gz)

Then:
disk-analyse -r 360:300 -S -s 0 -e 39 --format=ibm_mfm_dd  test.scp test.img
T0.0-9.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)
T10.0: IBM-MFM DD (8 512-byte sectors, 4096 bytes)
T11.0-22.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)
T23.0: IBM-MFM DD (8 512-byte sectors, 4096 bytes)
T24.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)
T25.0: IBM-MFM DD (8 512-byte sectors, 4096 bytes)
T26.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)
T27.0-28.0: IBM-MFM DD (8 512-byte sectors, 4096 bytes)
T29.0-39.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)

I get a sector image that's kinda, sorta the right size and appears to have at some of the data decoded.  No problem reading this disk on a PC, so not sure where to go from here.

**Reply by keirf** [reply]
Cross-checked by loading into HxC, it also sees some bad sectors on this disk. There are less than half dozen bad sectors, hence this disk is only 1-2% bad and would mostly work in a PC.

Some more magic parameters do a slightly better job:
```
viper:Downloads$ disk-analyse -r 360:300 -S -s 0 -e 39 --format=ibm_mfm_dd -p 1 -P 50 test.scp test.img
T0.0-9.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)
T10.0: IBM-MFM DD (8 512-byte sectors, 4096 bytes)
T11.0-24.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)
T25.0: IBM-MFM DD (8 512-byte sectors, 4096 bytes)
T26.0-27.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)
T28.0: IBM-MFM DD (8 512-byte sectors, 4096 bytes)
T29.0-39.0: IBM-MFM DD (9 512-byte sectors, 4608 bytes)
```

The `pP` parameters are arcane, but you can have a play around with them and sometimes extract marginal sectors.

**Reply by keirf** [reply]
Here is a flux map of track 28.0. x axis is elapsed time, y axis is time between adjacent fluxes. You can see the five revolutions (red ticks are the index pulses). You can see the usual "bit bands" at roughly 3.2us, 5us, 6.8s, but there's a blurred section with no defined banding. No fancy bit extraction algorithm is going to be able to handle that area.

EDIT: `python3 Greaseweazle/scripts/misc/scp_info.py --plot test.scp 56`

![Untitled](https://user-images.githubusercontent.com/879766/109728531-478ed100-7bae-11eb-8503-17f77d1acb93.png)

**Reply by snhirsch** [reply]
Interesting.  To be sure I understand the flux map, the 'blurred section' is represented by the five vertical blue "smears"? I appreciate your taking the time to explain matters!

**Reply by keirf** [reply]
Yes, the five vertical smears that go fairly solidly top to bottom.You need white space between the horizontal bands, so that bit timings can be distinguished: 3.2us means bits 01, 5us means 001, 6.8us means 0001. These bitcodes are concatenated together according to successive flux timings, and that forms the full MFM-encoded data stream as recorded on the track.

**Comment by snhirsch** [comment]
Just wanted to follow up on this.  I switched to a 360k DSDD drive and all the problems disappeared.  I was able to make clean images of every diskette in the set!  I think the lesson here is that a read head the full width of the track does a much better job of snagging bits from old media.  I was hoping to have a single setup that could handle DD and HD, but clearly that has its drawbacks.
